src/Utils/Strict.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+/**
+ * This file is part of the Nette Framework (https://nette.org)
+ * Copyright (c) 2004 David Grudl (https://davidgrudl.com)
+ */
+
+namespace Nette;
+
+use Nette\Utils\ObjectMixin;
+
+
+/**
+ * Strict class for better experience.
+ * - 'did you mean' hints
+ * - access to undeclared members throws exceptions
+ */
+trait Strict
+{
+
+	/**
+	 * Call to undefined method.
+	 * @throws MemberAccessException
+	 */
+	public function __call($name, $args)
+	{
+		$class = method_exists($this, $name) ? 'parent' : get_class($this);
+		$hint = ObjectMixin::getSuggestion((new \ReflectionClass($this))->getMethods(\ReflectionMethod::IS_PUBLIC), $name);
+		throw new MemberAccessException("Call to undefined method $class::$name()" . ($hint ? ", did you mean $hint()?" : '.'));
+	}
+
+
+	/**
+	 * Call to undefined static method.
+	 * @throws MemberAccessException
+	 */
+	public static function __callStatic($name, $args)
+	{
+		$rc = new \ReflectionClass(get_called_class());
+		$hint = ObjectMixin::getSuggestion(array_intersect(
+			$rc->getMethods(\ReflectionMethod::IS_PUBLIC),
+			$rc->getMethods(\ReflectionMethod::IS_STATIC)
+		), $name);
+		throw new MemberAccessException("Call to undefined static method {$rc->getName()}::$name()" . ($hint ? ", did you mean $hint()?" : '.'));
+	}
+
+
+	/**
+	 * Access to undeclared property.
+	 * @throws MemberAccessException
+	 */
+	public function &__get($name)
```

Comment:
btw: Nette has inconsistent coding style of writing `&` before methods names, it would be nice to unify it.